File: Lab8/Question3/src/Shape.java
public abstract class Shape {
    private double x;
    private double y;

    public Shape(double x, double y) {
        this.x = x;
        this.y = y;
    }

    public double getX() {
        return x;
    }

    public double getY() {
        return y;
    }

    public void move(double dx, double dy) {
        x += dx;
        y += dy;
    }

    public abstract double area();

    public abstract void resize(double newSize) throws Exception;

    public static void testShape() {
        System.out.println("Shape is an abstract class and cannot be initialized.");
    }
}

File: Lab8/Shape.kt
abstract class Shape {
    private var x: Double
    private var y: Double

    constructor(x: Double, y: Double) {
        this.x = x
        this.y = y
    }

    fun getX(): Double {
        return x
    }

    fun getY(): Double {
        return y
    }

    fun move(dx: Double, dy: Double) {
        x += dx
        y += dy
    }

    abstract fun area(): Double

    abstract fun resize(newSize: Double)
}